src/ASM/LR/ASMu2Dnurbs.C

```diff
@@ -289,3 +289,183 @@ void ASMu2D::computeBasisNurbs (double u, double v,
     bas.basisDerivs_uvv[i] = (G2x*W - 3.0*G2*Wx)*w[i]/(W*W*W*W);
   }
 }
+
+void ASMu2D::writePostscriptElementsNurbs (std::shared_ptr<LR::LRSplineSurface> mesh,
+                                           std::ostream& out, bool close,
+                                           int nu, int nv)
+{
+  // get date
+  time_t t = time(0);
+  tm* lt = localtime(&t);
+  char date[128];
+  sprintf(date, "%02d/%02d/%04d", lt->tm_mday, lt->tm_mon + 1, lt->tm_year+1900);
+
+  if (mesh != lrspline) {
+    mesh.swap(lrspline);
+    geo = lrspline.get();
+  }
+
+  // This is done unconditionally as it will not have been performed
+  // for the refined mesh. It will be on the next adaptive cycle though,
+  // so we do it here instead of in general to avoid paying the price if not
+  // outputting eps files.
+  this->generateBezierBasis();
+  this->generateBezierExtraction();
+
+  // get bounding box (max/min of the control points)
+  std::array<double,2> x{1e7, -1e7};
+  std::array<double,2> y{1e7, -1e7};
+  for (const LR::Basisfunction* b : lrspline->getAllBasisfunctions()) {
+      std::vector<double>::const_iterator cp = b->cp();
+      x[0] = (cp[0] < x[0]) ? cp[0] : x[0];
```

Comment:
operator[] on an const_iterator is well-defined, I assume?

Reply:
yes it is.